src/main/java/com/google/jenkins/plugins/storage/StdoutUpload.java

```diff
@@ -102,7 +103,10 @@ protected UploadSpec getInclusions(AbstractBuild<?, ?> build,
       OutputStream outputStream = null;
       try {
         FilePath logDir = new FilePath(build.getLogFile()).getParent();
-        FilePath logFile = new FilePath(logDir, getLogName());
+
+        String resolvedLogName =
+            Util.replaceMacro(getLogName(), build.getEnvironment(listener));
+        FilePath logFile = new FilePath(logDir, resolvedLogName);
```

Comment:
Note that this sort of patch is unnecessary for Pipeline: the script can use GString interpolation without any special support from the plugin. [more](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#variable-substitutions)

Reply:
Thanks for pointing this out!
Currently the pipeline is unsupported, but I will make sure to hunt down these substitutions when we add pipeline compatibility (this plugin already has quite a number of other places where similar substitutions happen).